### PR TITLE
Add generating table describing models using torchinfo

### PR DESCRIPTION
**Pull request by sliwy**
It adds method and overwrites default `__str__` behavior to represent a model with table generated by torchinfo.summary.

Resolve [#344](https://github.com/braindecode/braindecode/issues/344)

For example, for `ShallowFBCSPNet` from `examples/model_building/plot_bcic_iv_2a_moabb_trial.py` it gives something like that:
```
============================================================================================================================================
Layer (type (var_name):depth-idx)        Input Shape               Output Shape              Param #                   Kernel Shape
============================================================================================================================================
ShallowFBCSPNet (ShallowFBCSPNet)        [1, 22, 125]              [1, 4]                    --                        --
├─Ensure4d (ensuredims): 1-1             [1, 22, 125]              [1, 22, 125, 1]           --                        --
├─Rearrange (dimshuffle): 1-2            [1, 22, 125, 1]           [1, 1, 125, 22]           --                        --
├─Conv2d (conv_time): 1-3                [1, 1, 125, 22]           [1, 40, 101, 22]          1,040                     [25, 1]
├─Conv2d (conv_spat): 1-4                [1, 40, 101, 22]          [1, 40, 101, 1]           35,200                    [1, 22]
├─BatchNorm2d (bnorm): 1-5               [1, 40, 101, 1]           [1, 40, 101, 1]           80                        --
├─Expression (conv_nonlin_exp): 1-6      [1, 40, 101, 1]           [1, 40, 101, 1]           --                        --
├─AvgPool2d (pool): 1-7                  [1, 40, 101, 1]           [1, 40, 2, 1]             --                        [75, 1]
├─Expression (pool_nonlin_exp): 1-8      [1, 40, 2, 1]             [1, 40, 2, 1]             --                        --
├─Dropout (drop): 1-9                    [1, 40, 2, 1]             [1, 40, 2, 1]             --                        --
├─Conv2d (conv_classifier): 1-10         [1, 40, 2, 1]             [1, 4, 1, 1]              324                       [2, 1]
├─LogSoftmax (softmax): 1-11             [1, 4, 1, 1]              [1, 4, 1, 1]              --                        --
├─Expression (squeeze): 1-12             [1, 4, 1, 1]              [1, 4]                    --                        --
============================================================================================================================================
Total params: 36,644
Trainable params: 36,644
Non-trainable params: 0
Total mult-adds (Units.MEGABYTES): 5.87
============================================================================================================================================
Input size (MB): 0.01
Forward/backward pass size (MB): 0.78
Params size (MB): 0.15
Estimated Total Size (MB): 0.93
============================================================================================================================================
```

### CODE REVIEW

**Comment by PierreGtch** [comment]
Nice @sliwy ! you should add `torchinfo` to `environment.yml` and to the list of dependencies in `setup.py`

File: examples/model_building/plot_bcic_iv_2a_moabb_cropped.py [hunk 1]
```diff
@@ -158,6 +158,9 @@
     final_conv_length=30,
 )
 
+# Display torchinfo table describing the model
+print(model)
```
Comment (PierreGtch) [inline examples/model_building/plot_bcic_iv_2a_moabb_cropped.py:162]: is it the __str__ or __repr__ displayed here?

File: requirements.txt [hunk 2]
```diff
@@ -7,4 +7,5 @@ skorch<0.14
 joblib
 mne
 einops
-docstring_inheritance
\ No newline at end of file
+docstring_inheritance
+torchinfo
```
Comment (PierreGtch) [inline requirements.txt:11]: Can you add this to `environment.yml` and to the list in `setup.py` for the CI?

**Review by PierreGtch — CHANGES_REQUESTED** [review]

File: examples/model_building/plot_bcic_iv_2a_moabb_cropped.py (hunk 1 above)
Comment (sliwy) [reply examples/model_building/plot_bcic_iv_2a_moabb_cropped.py:162]: it's gonna be `str` as print first tries the `__str__` method and if it does not exist that fallbacks to `__repr__`

File: requirements.txt (hunk 2 above)
Comment (sliwy) [reply requirements.txt:11]: done

**Comment by sliwy** [comment]
Thx @PierreGtch 3 places to add the package, way too many for me to remember.

Are we checking anywhere that conda environment is built properly? I am not using conda, so you never know... :wink: torchinfo should be accessible through conda-forge, so should be fine I guess.

File: examples/model_building/plot_bcic_iv_2a_moabb_cropped.py (hunk 1 above)
Comment (sliwy) [reply examples/model_building/plot_bcic_iv_2a_moabb_cropped.py:162]: we may call `str` explicitly in case you feel it would be easier to read

**Comment by PierreGtch** [comment]
> Thx @PierreGtch 3 places to add the package, way too many for me to remember.
> 
> Are we checking anywhere that conda environment is built properly? I am not using conda, so you never know... 😉 torchinfo should be accessible through conda-forge, so should be fine I guess.

No I don’t think the requirements and the conda environment are tested. I agree, it’s not the best...

**Comment by PierreGtch** [comment]
@sliwy you need to fix the syntax to be compatible with python 3.8 (check the CI tests)

**Comment by sliwy** [comment]
done, just rerun the CI :)

**Comment by sliwy** [comment]
I have fixed the flake8, and there was some strange test fail, looking like it's coming from CI misconfiguration. Let's rerun and hope it works now :)

**Review by PierreGtch — APPROVED** [review]

**Pull request merged**